File: src/main/java/de/danielmaile/aether/util/serializer/VectorSerializer.java
package de.danielmaile.aether.util.serializer;

import com.google.gson.*;
import org.bukkit.util.Vector;

import java.lang.reflect.Type;

public class VectorSerializer implements JsonSerializer<Vector>, JsonDeserializer<Vector>
{
    @Override
    public JsonElement serialize(Vector vector, Type type, JsonSerializationContext jsonSerializationContext)
    {
        JsonObject jsonObject = new JsonObject();
        jsonObject.addProperty("x", vector.getX());
        jsonObject.addProperty("y", vector.getY());
        jsonObject.addProperty("z", vector.getZ());
        return jsonObject;
    }

    @Override
    public Vector deserialize(JsonElement jsonElement, Type type, JsonDeserializationContext jsonDeserializationContext) throws JsonParseException
    {
        JsonObject jsonObject = jsonElement.getAsJsonObject();
        double x = jsonObject.get("x").getAsDouble();
        double y = jsonObject.get("y").getAsDouble();
        double z = jsonObject.get("z").getAsDouble();
        return new Vector(x, y, z);
    }
}

File: src/main/java/de/danielmaile/mpp/util/serializer/VectorSerializer.kt
package de.danielmaile.mpp.util.serializer

import com.google.gson.*
import kotlin.Throws
import org.bukkit.util.Vector
import java.lang.reflect.Type

class VectorSerializer : JsonSerializer<Vector>, JsonDeserializer<Vector> {

    override fun serialize(
        vector: Vector,
        type: Type,
        jsonSerializationContext: JsonSerializationContext
    ): JsonElement {
        val jsonObject = JsonObject()
        jsonObject.addProperty("x", vector.x)
        jsonObject.addProperty("y", vector.y)
        jsonObject.addProperty("z", vector.z)
        return jsonObject
    }

    @Throws(JsonParseException::class)
    override fun deserialize(
        jsonElement: JsonElement,
        type: Type,
        jsonDeserializationContext: JsonDeserializationContext
    ): Vector {
        val jsonObject = jsonElement.asJsonObject
        val x = jsonObject["x"].asDouble
        val y = jsonObject["y"].asDouble
        val z = jsonObject["z"].asDouble
        return Vector(x, y, z)
    }
}